### ISSUE TITLE
MSE HLS?

### ISSUE THREAD

**Issue by heff** [videojs/videojs-contrib-hls#68]
Just hoping to get an idea of how far we are from supporting HLS through MSE. That may be something I try to help out with here soon if it's needed.

We talked about this briefly before, but one thing I'm wondering is if we'll end up with two techs, hls-mse and hls-flash, since the current HLS extends vjs.Flash, and the easiest way to add MSE HLS would seem to be to extend vjs.Html5, like was done in [vjs.Dashjs](https://github.com/heff/videojs-dashjs).

**Comment by sabberworm** [comment]
@gkatsev Regarding the “downside” you mentioned of having to declare hls three times in `techOrder`: I, for one, would love that. I have found hls-flash to be incredibly buggy and would love for there to be a way to excluding it from the available techs whilst keeping native and mse in.

**Comment by sabberworm** [comment]
So if this issue was closed because the “design issue is resolved”, should I open a new issue for actually implementing HLS via MSE?

**Comment by dmlap** [comment]
@sabberworm we're *right* in the middle of it, so no need! If you'd like to check out our progress, you can keep an eye on the [mse-magic](https://github.com/dmlap/videojs-contrib-hls/tree/mse-magic) branch in my fork. I'll be opening a PR on the main repo as soon as I get all the tests passing again.

**Comment by sabberworm** [comment]
@dmlap That’s great news!

**Comment by arikgilad** [comment]
I tried to test the mse-magic version but I was not able to make it work.
When running the example I found many issues:
1. in videojs-hls.js i got typerror on 
videojs.Hls = videojs.extends(Component, {...

I had to change videojs.extends to videojs.extend

2. The code is using request.response/request.responseText in videojs-hls.js and playlist-loader.js instead of request.body.

Even after fixing those issues, the video was still not playing.
It was downloading all the ts files but never played them because drainBuffer() was doing nothing because this.sourceBuffer.updating was set once and never unset.
